src/std/parser/defparser.ss

```diff
@@ -130,7 +131,7 @@ package: std/parser
            (reverse prods))))))
 
   (def (generate-prod-id base part)
-    (datum->syntax base (make-symbol (stx-e base) "::" part)))
+    (format-id base "~a::~a" (stx-e base) part))
```

Comment:
the `stx-e` on base is unnecessary, `format-id` will do it automatically.

Reply:
Do you want to keep `generate-prod-id` or refactor completely to use `format-id` everywhere?